### DISCUSSION TITLE
Markdoc How?

### DISCUSSION THREAD

**Discussion by rawnly** [remix-run/remix#3350] [Q&A] [answered]
Hello everyone! I'm trying to move my nextjs blog to remix, the only problem I'm encountering right now is that I currently use [markdoc][markdoc] to parse my files and I can't find a way to integrate it in remix. I've read all the mdx stuff in the docs but as far as I understood i'm limited to remark.

Does anyone have any idea? (maybe creating a custom remark plugin.. )

[markdoc]: https://markdoc.io

**Comment by Xiphe** [comment, marked as answer]
If you want to [use `.mdx` files as routes](https://remix.run/docs/en/v1/guides/mdx#routes) you'll probably need to convert your files to be parsable by remark.

But you still can roll your own implementation. 
I'd recommend checking out the blog implementation of [kentcdodds.com](https://github.com/kentcdodds/kentcdodds.com)  for a production reference

It's served from [.md files on github](https://github.com/kentcdodds/kentcdodds.com/tree/main/content/blog).
See [route implementation](https://github.com/kentcdodds/kentcdodds.com/blob/main/app/routes/blog.%24slug.tsx) + the underlying markdown handlers [mdx.ts](https://github.com/kentcdodds/kentcdodds.com/blob/main/app/utils/mdx.tsx#L40e) and [compile-mdx.server.ts](https://github.com/kentcdodds/kentcdodds.com/blob/main/app/utils/compile-mdx.server.ts) ...they still use remark here, but a similar approach will also work with all other sorts of just-in-time content preparation.

**Reply by rawnly** [reply]
Thanks! I'll have a look

**Reply by niklasgrewe** [reply]
@Rawnly have you found a solution yet? I have the same problem 😅

**Reply by rawnly** [reply]
@niklasgrewe no I'm sorry but paused that project lol
